src/zjs_modules.c

```diff
@@ -32,28 +33,27 @@ struct routine_map {
 static u8_t num_routines = 0;
 struct routine_map svc_routine_map[NUM_SERVICE_ROUTINES];
 
+    static jerryx_module_resolver_t resolvers[1] =
+    {
+        jerryx_module_native_resolver
+    };
+
 static ZJS_DECL_FUNC(native_require_handler)
 {
     // args: module name
     ZJS_VALIDATE_ARGS(Z_STRING);
 
     jerry_size_t size = MAX_MODULE_STR_LEN;
-    char module[size];
+    jerry_char_t module[size];
     zjs_copy_jstring(argv[0], module, &size);
     if (!size) {
         return RANGE_ERROR("argument too long");
     }
 
-    int modcount = sizeof(zjs_modules_array) / sizeof(module_t);
-    for (int i = 0; i < modcount; i++) {
-        module_t *mod = &zjs_modules_array[i];
-        if (strequal(mod->name, module)) {
-            // We only want one instance of each module at a time
-            if (mod->instance == 0) {
-                mod->instance = mod->init();
-            }
-            return jerry_acquire_value(mod->instance);
-        }
+    jerry_value_t js_module = jerryx_module_resolve (module, resolvers, 1);
+    // If we found our module, return it.
+    if (!jerry_value_has_error_flag (js_module)) {
+        return js_module;
     }
```

Comment:
If it *does* have the error flag we need to `jerry_release_value (js_module);` otherwise we leak the error in `js_module`.

Reply:
Updated thanks.